muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/main-function/users/index.ts

```diff
@@ -70,30 +83,53 @@ let applyStudent: ApplyStudentTriggerType = function applyStudent (data) {
   }
 }
 
-const selectStaffmemberEndPoint:  ApplyStaffmemberTriggerType = function studentEndPoint (data) {
-  if(data.mode  == "CREATE" ) {
-    return mApi().user.staffMembers.create(data.staffmember);
-  } else {
-    return mApi().user.staffMembers.put(data.staffmember);
+let updateStudent: UpdateStudentTriggerType = function updateStudent(data) {
+  return async (dispatch: (arg: AnyActionType) => any, getState: () => StateType) => {
+    try {
+      await promisify(mApi().user.students.basicInfo.update(data.student.identifier, data.student), 'callback')().then(() => {
+
+        mApi().organizationUserManagement.students.cacheClear();
+
+        setTimeout(async () => {
+          let users: UsersListType = await promisify(mApi().organizationUserManagement.students.read(), 'callback')() as UsersListType;
+          dispatch({
+            type: "UPDATE_STUDENT_USERS",
+            payload: users
+          });
+        }, 2000);
+      });
+      dispatch(notificationActions.displayNotification(getState().i18n.text.get("plugin.organization.update.student.success"), 'success'));
+      data.success && data.success();
+    } catch (err) {
+      if (!(err instanceof MApiError)) {
+        throw err;
+      }
+      dispatch(notificationActions.displayNotification(getState().i18n.text.get("plugin.organization.update.student.error"), 'error'));
+      data.fail && data.fail();
+    }
   }
 }
 
-let applyStaffmember: ApplyStaffmemberTriggerType = function applyStaffmember (data) {
-  return async (dispatch:(arg:AnyActionType)=>any, getState:()=>StateType)=>{
+let createStaffmember: CreateStaffmemberTriggerType = function createStaffmember(data) {
+  return async (dispatch: (arg: AnyActionType) => any, getState: () => StateType) => {
     try {
-      await promisify(selectStaffmemberEndPoint(data), 'callback')().then(() => {
-        promisify(mApi().user.staffMembers.read(), 'callback')()
-        .then((users:UserType)=>{
+      await promisify(mApi().user.staffMembers.create(data.staffmember), 'callback')().then(() => {
+
+        mApi().organizationUserManagement.staffMembers.cacheClear();
+
+        setTimeout(async () => {
```

Comment:
Because create : muikku --> pyramus ---> muikku --> and if it just refreshes, it's too fast and no changes in state. I made it 2 second just to be sure.